File: packages/agent-core/src/thread.ts
import type { ThreadId, OrgId, UserId, WorkspaceId, SkillId } from "./ids.js";

export type AgentThreadStatus = "active" | "archived";

export type AgentThread = {
  id: ThreadId;
  orgId: OrgId;
  ownerUserId: UserId;
  title: string;
  status: AgentThreadStatus;
  workspaceId: WorkspaceId;
  defaultSkillId?: SkillId;
  createdAt: string;
  updatedAt: string;
};

File: backend/src/aithru_agent/domain/thread.py
from enum import StrEnum

from .base import AithruBaseModel


class AgentThreadStatus(StrEnum):
    ACTIVE = "active"
    ARCHIVED = "archived"


class AgentThread(AithruBaseModel):
    id: str
    org_id: str
    owner_user_id: str
    title: str | None = None
    status: AgentThreadStatus = AgentThreadStatus.ACTIVE
    created_at: str
    updated_at: str

